src/main.py

```diff
@@ -66,8 +67,69 @@ async def start(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
     )
 
 
+# TODO: combine common logic for add_team and remove_team into one method
 async def add_team(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
-    pass
+    if not context.args:
+        await update.message.reply_text(
+            "Team_id parameter is required. Usage: /add_team <team_id>"
+        )
+        return
+
+    chat_id = update.message.chat_id
+    club = chat_registry.get_club(chat_id)
+    if not club:
+        await update.message.reply_text(
+            f"The club with id #{chat_id} does not exist, please register it first by using /start command"
+        )
+        return
+
+    if club.has_team(context.args[0]):
+        await update.message.reply_text(
+            f"Team with id {context.args[0]} already exists in the club with chat_id: {chat_id}"
+        )
+        return
+
+    # TODO: for now TeamBase is used
+    # TODO: get Team name from the system
+    club.add_team(
+        TeamBase(
+            context.args[0],
+            update.message.message_thread_id,
+            f"Team {update.message.message_thread_id}",
+        )
+    )
+    chat_registry.update_club(club)
+    await update.message.reply_text(
+        f"Team with id {context.args[0]} has been added to the club with chat_id: {chat_id}"
+    )
+
+
+async def remove_team(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    if not context.args:
+        await update.message.reply_text(
+            "The team_id argument is required. Usage: /remove_team <team_id>"
+        )
+        return
+
+    chat_id = update.message.chat_id
+    club = chat_registry.get_club(chat_id)
+    if not club:
+        await update.message.reply_text(
+            f"The club with id #{chat_id} does not exist, please register it first by using /start command"
+        )
+        return
+
+    if not club.has_team(context.args[0]):
+        await update.message.reply_text(
+            f"Team with id {context.args[0]} does not exist in the club with chat_id: {chat_id}"
+        )
+        return
+
+    club.remove_team(context.args[0])
+    chat_registry.update_club(club)
```

Comment:
Instead of such straightforward logic, I would suggest setting a flag indicating that the club/team needs to be updated. Then, in the regular task of ChatRegistry, it would perform the restoration xD. What do you think about this proposal? Ready to discuss

Benefits exclude this logic from main, prevent concurrent access to the data. Ideally using mutexes at the time of  adding\removing\unloading data.